test/test_unistd.rs

```diff
@@ -98,6 +98,41 @@ fn test_mkfifo_directory() {
     assert!(mkfifo(&env::temp_dir(), Mode::S_IRUSR).is_err());
 }
 
+#[test]
+fn test_mkfifoat() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let mkfifoat_fifo = tempdir.path().join("mkfifoat_fifo");
+
+    mkfifoat(None, &mkfifoat_fifo, Mode::S_IRUSR).unwrap();
+
+    let stats = stat::stat(&mkfifoat_fifo).unwrap();
+    let typ = stat::SFlag::from_bits_truncate(stats.st_mode);
+    assert!(typ == SFlag::S_IFIFO);
+
+
+    let dirfd = open(tempdir.path(), OFlag::empty(), Mode::empty()).unwrap();
```

Comment:
This part should really be a separate test case.

Reply:
fixed